src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -1958,11 +1960,47 @@ void UpdatePeriodicBackups()
                     if (IsExcluded(configurationBlittable, addDatabaseCommand.Name))
                         continue;
 
+                    configurationBlittable.TryGet(nameof(ServerWideBackupConfiguration.BackupType), out BackupType backupType);
+                    if (addDatabaseCommand.Record.IsSharded && backupType == BackupType.Snapshot)
+                    {
+                        // sharded database are excluded from server-wide snapshot backups
+                        // add the database name to 'ExcludedDatabases' list
+
+                        serverWideBackups.Modifications ??= new DynamicJsonValue(serverWideBackups);
+                        configurationBlittable.Modifications ??= new DynamicJsonValue(configurationBlittable);
+
+                        if (configurationBlittable.TryGet(nameof(IServerWideTask.ExcludedDatabases), out BlittableJsonReaderArray excludedDatabases))
+                        {
+                            excludedDatabases.Modifications ??= new DynamicJsonArray();
+                            excludedDatabases.Modifications.Add(addDatabaseCommand.Name);
+                        }
+                        else
+                        {
+                            configurationBlittable.Modifications[nameof(IServerWideTask.ExcludedDatabases)] =
+                                new DynamicJsonArray(new[] { addDatabaseCommand.Name });
+                        }
+
+                        serverWideBackups.Modifications[propertyName] = configurationBlittable;
+                        shouldUpdateServerWideBackups = true;
+
+                        continue;
+                    }
+
                     var backupConfiguration = JsonDeserializationCluster.PeriodicBackupConfiguration(configurationBlittable);
                     PutServerWideBackupConfigurationCommand.UpdateTemplateForDatabase(backupConfiguration, addDatabaseCommand.Name, addDatabaseCommand.Encrypted);
                     addDatabaseCommand.Record.PeriodicBackups.Add(backupConfiguration);
                     hasChanges = true;
                 }
+
+                if (shouldUpdateServerWideBackups)
+                {
+                    using (serverWideBackups)
+                    {
+                        serverWideBackups = context.ReadObject(serverWideBackups, ServerWideConfigurationKey.Backup);
```

Comment:
I know the code is fine, but please fix those warning :)

Reply:
fixed